caikit/core/data_model/base.py

```diff
@@ -171,6 +181,21 @@ def parse_proto_descriptor(mcs, cls):
             if field.enum_type is not None
         }
 
+        # mapping of all oneofs and the fields that are part of them
+        # NOTE: protobuf makes an interesting use of oneof to wrap types that
+        #   should be explicitly optional. We don't want to consider these
+        #   oneofs in the general oneof handling.
+        cls._fields_oneofs_map = {
+            oneof_name: [field.name for field in oneof.fields]
+            for oneof_name, oneof in cls._proto_class.DESCRIPTOR.oneofs_by_name.items()
+            if len(oneof.fields) != 1 or oneof.name != f"_{oneof.fields[0].name}"
```

Comment:
Yep, this is just trying to identify the entries in `oneofs` that are only there for `optional` fields and removing them from the list

Reply:
To make sure I understood this correctly - 
```
message Foo {
    int32 bar = 1;
    optional int32 baz = 2;
}
``` 
becomes
```
message Foo {
    int32 bar = 1;
    oneof baz {
        int32 baz = 2;
    }
}
```
because then `oneof.name` is `baz`, which is the same as `oneof.fields[0].name`?

Reply:
This is close, but the name of the oneof is `_baz` not `baz`